### PR TITLE
feat: duplicate upstream service consumer

### LINKED ISSUE #1487: Support duplicate one existing Route
# Feature request

## Please describe your feature

Support duplicate one existing Route

## Describe the solution you'd like

1. Add a Duplicate button on the Route List page;
2. When clicking the Duplicate button, we should first fetch that route's data, then call Create a Route method to implement this feature.

### PR DESCRIPTION
**Pull request by batman-ezio**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Add duplicate feature to upstream, service and consumer, just like the feature in route

**Related issues**

fix/resolve #1487 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
wow! good

**Comment by juzhiyuan** [comment]
also cc @guoqqqi @iamayushdas @KishaniKandasamy and more contributors to review :)

**Comment by nic-chen** [comment]
Conflicting, please resolve it , thanks.

File: web/src/pages/Upstream/Create.tsx [hunk 1]
```diff
@@ -30,13 +30,19 @@ const Page: React.FC = (props) => {
...
 
   useEffect(() => {
     const { id } = (props as any).match.params;
 
     if (id) {
       fetchOne(id).then((data) => {
         form1.setFieldsValue(data.data);
+        if (isDuplicate) {
+          form1.setFieldsValue({
+            'name': ''
```
Comment (LiteSun) [inline web/src/pages/Upstream/Create.tsx:43]:
```suggestion
            name: ''
```

File: web/cypress/integration/consumer/create-edit-duplicate-and-delete-consumer.spec.js [hunk 2]
```diff
@@ -58,6 +58,70 @@ context('Create and Delete Consumer', () => {
...
+
+  });
+
+  it('should duplicate the consumer', function () {
+    cy.visit('/consumer/list');
+
+    cy.contains('Search').click();
+    cy.get(this.domSelector.nameSelector).type(this.data.consumerName);
+    cy.contains(this.data.consumerName).siblings().contains('Duplicate').click();
+    cy.get(this.domSelector.username).type(this.data.consumerName2);
```
Comment (guoqqqi) [inline web/cypress/integration/consumer/create-edit-duplicate-and-delete-consumer.spec.js:99]:
```suggestion
    cy.get(this.domSelector.username).should('not.have.value').type(this.data.consumerName2);
```

**Comment by guoqqqi** [comment]
Hi, there are some conflicting files that need to be resolved.

File: web/src/pages/Upstream/List.tsx [hunk 3]
```diff
@@ -92,6 +92,12 @@ const Page: React.FC = () => {
           }}>
             {formatMessage({ id: 'component.global.view' })}
           </Button>
+          <Button
+            type="primary"
+            onClick={() => history.push(`/upstream/${record.id}/duplicate`)}
+          >
+            {formatMessage({ id: 'page.upstream.list.duplicate' })}
```
Comment (bzp2010) [inline web/src/pages/Upstream/List.tsx:99]:
```suggestion
            {formatMessage({ id: 'component.global.duplicate' })}
```

File: web/src/pages/Upstream/locales/en-US.ts [hunk 4]
```diff
@@ -88,6 +88,7 @@ export default {
   'page.upstream.list.edit.time': 'Updated At',
   'page.upstream.list.operation': 'Operation',
   'page.upstream.list.edit': 'Configure',
+  'page.upstream.list.duplicate': 'Duplicate',
```
Comment (bzp2010) [inline web/src/pages/Upstream/locales/en-US.ts:91]:
```suggestion

```

File: web/src/pages/Upstream/locales/zh-CN.ts [hunk 5]
```diff
@@ -88,6 +88,7 @@ export default {
   'page.upstream.list.edit.time': '更新时间',
   'page.upstream.list.operation': '操作',
   'page.upstream.list.edit': '配置',
+  'page.upstream.list.duplicate': '复制',
```
[the remark this answers is not in the document]
Comment (bzp2010) [reply web/src/pages/Upstream/locales/zh-CN.ts:91]:
```suggestion

```

**Pull request closed without merging**